### DISCUSSION TITLE
Other attempts for TorchSharp RL

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [asieradzk/RL_Matrix#2] [General]
[First WinForm .NET6 TorchSharp QLearning Example](https://github.com/xin-pu/DeepSharp/discussions/10)

It is interesting that this project is providing interesting UI visualization for displaying Deep RL training progress and 2D display.

Hopefully, others would like to join to contribute this project and add 3D visualization using e.g. **Godot** and charting training using **Tensorboard** as part of Torchsharp

=================

Do consider **porting** this legacy Winform DeepQLearning to Torchsharp and .NET6/7

- [Github](https://github.com/dubezOniner/Deep-QLearning-Demo-csharp/tree/master/DeepQLearning)
- [Web](https://cs.stanford.edu/people/karpathy/convnetjs/demo/rldemo.html)
- [Youtube](https://www.youtube.com/watch?v=pTgI_-yYlBA&t=9s)
![Items](https://github.com/xin-pu/DeepSharp/assets/49812372/93ca452b-9d8d-497f-a912-8465bacb0bdd)

**Comment by asieradzk** [comment]
Thanks for sharing, this is indeed cool.

As for 2d/3d visualisation, next on my checklist is making a Unity implementation to rule over Unity's ML agents (which have very cool way to setup a scene but huge drawback of spinning up TensorFlow session in the background and communicating over local socket) 

Did you see it?

https://unity.com/products/machine-learning-agents


But yeah at any point later I dont see why I shouldn't create a sub namespace/separate nuget that will include this project :)

**Reply by GeorgeS2019** [reply]
I have done the feasibility studies, there are Godot RL Agent, and other attempts.

The key is to make users come from.NET6 (which is still not available YET in Unity, but existed in Godot since 2022) to do the same as Unity ML Agent But on the top 8th Ranked FOSS ( Free Open Source Software with MIT license ) => Godot with fast short learning curve and many other advantages

**Reply by GeorgeS2019** [reply]
If you are interested for more people joining this project, especially team members from Dotnet foundation, you need to have a **MIT** license. 

Do not worry about not citing you. 

There are still many many many steps toward commercialization and others will recruit you to their industry applications

**Reply by asieradzk** [reply]
-I understand, so .NET6 is not available in Unity (and will probably never be)
-Thank you so much for the exciting news about Godot, I will check out how its done. So you'd like to have a .NET6 deep reinforcement learning agents similar to ML-Agents? 

Didn't know about the license considerations - as I am quite new to open source, I'll check with my senior contemporaries and see about change later:
https://www.youtube.com/watch?v=Vhh_GeBPOhs
